receiver/mongodbreceiver/scraper_test.go

```diff
@@ -288,7 +229,7 @@ func TestScraperScrape(t *testing.T) {
 				require.NoError(t, err)
 				return expectedMetrics
 			},
-			expectedErr: nil,
+			expectedErr: errLocksPartialMetrics,
```

Comment:
Why can't we test these metrics?

Reply:
We do test these metrics. Some of the lock metrics added in this PR are not necessarily published at all times. 

MongoDB does not exclude lock related metrics if the metric value is 0 in its serverStatus diagnostic calls. Accordingly, the successful scrap test case would have partial errors and the test case checks for partial error related to absent lock metrics.

Reply:
Do you have any thoughts on how to deal with such metrics? As we add new metrics to mongodb receiver we will have to deal with metrics which are not always published

Reply:
I think it is not an error then if we expect it will not be present in some circumstances.

Reply:
Agreed, modifying code to not raise a error when metrics which are not present and are not expected to be always published

Reply:
Agreed, I am updating the code to not raise an error when a metric key is absent and the metric is not expected to be present at all times.